### ISSUE TITLE
[bug] `@eslint-react/no-unnecessary-use-ref`: false positive when storing previous value in ref

### ISSUE THREAD

**Issue by ulrichstark** [Rel1cx/eslint-react#1406]
### Describe the bug

We use React refs to store the value of previous renders and to act accordingly in effects.

### Reproduction

```ts
import { useEffect, useRef } from "react";

export function useBooleanTrigger(
  value: boolean,
  { turnedTrue, turnedFalse }: { turnedTrue?: () => void; turnedFalse?: () => void },
) {
  const previousValueRef = useRef(value);

  useEffect(() => {
    const { current: previousValue } = previousValueRef;

    if (value !== previousValue) {
      previousValueRef.current = value;

      if (value) {
        turnedTrue?.();
      } else {
        turnedFalse?.();
      }
    }
  }, [turnedFalse, turnedTrue, value]);
}
```

### Expected behavior

Don't report the use of the ref, because we need it to store the value of a previous render.

### Platform and versions

```Text
@eslint-react/eslint-plugin: 2.6.4
```

### Stack trace

```Text

```

### Additional context

There may be a way to structure this code differently to fix the lint reporting and simultaneously preserve behaviour, but I can't think of one.
